opentelemetry-exporter-geneva/geneva-uploader/src/ingestion_service/mod.rs

```diff
@@ -0,0 +1,242 @@
+            std::env::var("GENEVA_CERT_PATH").expect("GENEVA_CERT_PATH is required"),
+        );
+        let cert_password = env::var("GENEVA_CERT_PASSWORD").unwrap_or_default();
+        let config_major_version = env::var("GENEVA_CONFIG_MAJOR_VERSION")
+            .expect("GENEVA_CONFIG_MAJOR_VERSION is required")
+            .parse::<u32>()
+            .expect("GENEVA_CONFIG_MAJOR_VERSION must be a u32");
+        let source_identity = env::var("GENEVA_SOURCE_IDENTITY")
+            .unwrap_or_else(|_| "Tenant=Default/Role=Uploader/RoleInstance=devhost".to_string());
+        let schema_ids =
+            "c1ce0ecea020359624c493bbe97f9e80;0da22cabbee419e000541a5eda732eb3".to_string(); //TODO make this configurable
+
+        let uploader_config = GenevaUploaderConfig {
+            namespace: namespace.clone(),
+            source_identity,
+            environment: environment.clone(),
+            schema_ids,
+        };
+
+        let config = GenevaConfigClientConfig {
+            endpoint,
+            environment,
+            account,
+            namespace,
+            region,
+            config_major_version,
+            auth_method: AuthMethod::Certificate {
+                path: cert_path,
+                password: cert_password,
+            },
+        };
+
+        let config_client =
+            GenevaConfigClient::new(config).expect("Failed to create config client");
+        let uploader = GenevaUploader::from_config_client(&config_client, uploader_config)
+            .await
+            .expect("Failed to create uploader");
+
+        let event_name = "Log";
+        let event_version = "Ver2v0";
+
+        println!("🚀 Launching {parallel_uploads} parallel uploads...");
+
+        let start_all = Instant::now();
+
+        let mut handles = vec![];
+        for i in 0..parallel_uploads {
+            let uploader = uploader.clone();
+            let data = data.clone();
+            let event_name = event_name.to_string();
+            let event_version = event_version.to_string();
+
+            let handle = tokio::spawn(async move {
+                let start = Instant::now();
+                match uploader.upload(data, &event_name, &event_version).await {
+                    Ok(resp) => {
+                        let elapsed = start.elapsed();
+                        println!(
+                            "✅ Upload {} complete in {:.2?}. Ticket: {}",
+                            i, elapsed, resp.ticket
+                        );
+                        Some(elapsed)
+                    }
+                    Err(e) => {
+                        eprintln!("❌ Upload {} failed: {:?}", i, e);
+                        None
+                    }
+                }
+            });
+            handles.push(handle);
+        }
+
+        let durations: Vec<_> = futures::future::join_all(handles)
+            .await
+            .into_iter()
+            .filter_map(|res| res.ok().flatten())
```

Comment:
Do we not want to ensure that we got a success response for every upload here? If not, let's add the number of uploads for which the average is calculated.